nomad/rpc_test.go

```diff
@@ -1274,18 +1274,22 @@ func TestRPC_TLS_Enforcement_RPC(t *testing.T) {
 			}
 
 			for method, arg := range tc.rpcs {
-				t.Run(fmt.Sprintf("nomad RPC: rpc=%s verify_hostname=true", method), func(t *testing.T) {
-					err := tlsHelper.nomadRPC(t, tlsHelper.mtlsServer, cfg, method, arg)
-
-					if tc.canRPC {
-						if err != nil {
-							require.NotContains(t, err, "certificate")
+				for _, srv := range []*Server{tlsHelper.mtlsServer1, tlsHelper.mtlsServer2} {
```

Comment:
This tests problem `2` by using a two server cluster and making requests to both. One of the requests will need to be forwarded.